### PR TITLE
Bump `eslint-plugin-import`,`@typescript-eslint/**`, and `downlevel-dts` and add rules for consistent type imports/exports

### PR DESCRIPTION
**Pull request by laurkim**
### WHY are these changes introduced?

Bumps `eslint-plugin-import` and `@typescript-eslint/**` packages to latest versions.
Adds `@typescript-eslint/consistent-type-imports`, `@typescript-eslint/consistent-type-exports`, and `import/consistent-type-specifier-style` rules and updates type imports to unblock v11 work.

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by BPScott — APPROVED** [review]
Big old pile of autofix changes look good to me

You might also be interested in the [`@typescript-eslint/no-import-type-side-effects`](https://typescript-eslint.io/rules/no-import-type-side-effects) and [`import/consistent-type-specifier`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/consistent-type-specifier-style.md) rules (I think you'll need a more up-to-date version of the `@typescript-eslint` packages and `eslint-plugin-import` than what's currently in the yarn.lock for those).

**Comment by laurkim** [comment]
> You might also be interested in the [`@typescript-eslint/no-import-type-side-effects`](https://typescript-eslint.io/rules/no-import-type-side-effects) and [`import/consistent-type-specifier`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/consistent-type-specifier-style.md) rules (I think you'll need a more up-to-date version of the `@typescript-eslint` packages and `eslint-plugin-import` than what's currently in the yarn.lock for those).

@BPScott I tried bumping the `@shopify/eslint-plugin` package (v42.0.3) to upgrade the `eslint-plugin-import` package (v2.26.0) but it looks like the `import/consistent-type-specifier-style` rule isn't available until [v2.27+](https://www.npmjs.com/package/eslint-plugin-import/v/2.27.4). 

I'll still add a commit to this PR to bump the `@shopify/eslint-plugin` pkg and add the `@typescript-eslint/no-import-type-side-effects` rule. I'll make a follow up PR to add the `import/consistent-type-specifier-style` rule once `eslint-plugin-import` is bumped in the next release of `@shopify/eslint-plugin`.

**Review by aveline — APPROVED** [review]

**Comment by BPScott** [comment]
> @BPScott I tried bumping the @shopify/eslint-plugin package (v42.0.3) to upgrade the eslint-plugin-import package (v2.26.0) but it looks like the import/consistent-type-specifier-style rule isn't available until [v2.27+](https://www.npmjs.com/package/eslint-plugin-import/v/2.27.4).


`@shopify/eslint-plugin` specifies its dependency on eslint-plugin-import with a caret. The existing version that polaris uses was `^2.25.3`. This version range shall accept any installed version as long as it starts with a `2.`. That includes 2.27.0. There is no update to `@shopify/eslint-plugin` required here because it already says that consuming apps are free to use 2.x.x versions of eslint-plugin-import. Polaris just needs to update the `eslint-plugin-import` version within its yarn.lock.

I don't understand what you've done to this yarn lock. You've introduced a large pile of duplicate dependencies but I can't see the reason why. All that needed to happen in this PR was:

- Delete mentions of `@typescript-eslint/*` packages and the `eslint-plugin-import` package in the yarn.lock
- Run `yarn` to regenerate the versions of these packages which shall pull in the latest versions of them that satisifies the ranges requested by `@shopify/eslint-plugin` - which in the case of eslint-plugin-import the range is `^2.25.3`,  whatever the latest 2.x version would be installed - which looks like 2.27.5.

File: .eslintrc.js [hunk 1]
```diff
@@ -50,6 +51,7 @@ module.exports = {
       },
     ],
     '@babel/no-unused-expressions': 'off',
+    'import/consistent-type-specifier-style': 'error',
```
Comment (aaronccasanova) [inline .eslintrc.js]:
```suggestion
    'import/consistent-type-specifier-style': ['error', 'prefer-top-level'],
```

Unlike the following rules, [import/consistent-type-specifier-style](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/consistent-type-specifier-style.md#options) defaults to `prefer-inline`

File: .eslintrc.js [hunk 2]
```diff
@@ -60,7 +62,10 @@ module.exports = {
     '@shopify/no-ancestor-directory-import': 'error',
     '@shopify/react-prefer-private-members': 'off',
     '@typescript-eslint/array-type': ['error', {default: 'array'}],
+    '@typescript-eslint/consistent-type-imports': 'error',
+    '@typescript-eslint/consistent-type-exports': 'error',
     '@typescript-eslint/naming-convention': 'off',
+    '@typescript-eslint/no-import-type-side-effects': 'error',
```
Comment (aaronccasanova) [inline .eslintrc.js]:
```suggestion
```

This rule can be removed with the above suggestion as inline `type` qualifiers will no longer be permitted in the repo

**Review by aaronccasanova — APPROVED** [review]
Left a couple cleanup suggestions, but the changes look great overall 🚀

**Pull request merged**